src/main/java/pl/bajerska/befindyourmeal/user/UserServiceImpl.java

```diff
@@ -1,30 +1,53 @@
 package pl.bajerska.befindyourmeal.user;
 
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
+import java.util.regex.Pattern;
+
 @Service
 @Transactional
 public class UserServiceImpl implements UserService {
 
-    @Autowired
     private UserRepository userRepository;
+    private BCryptPasswordEncoder bCryptPasswordEncoder;
+
+    @Autowired
```

Comment:
I'm not 100% sure, but probably it can work even without this annotation (some SpringBoot magic). But please check. :)